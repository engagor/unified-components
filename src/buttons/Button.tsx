import '@clarabridge/unified-ui/dist/buttons.css';
import React, { PropsWithChildren, ReactElement } from 'react';

export type ButtonProps = {
    mode: 'default' | 'primary' | 'secondary' | 'danger';
    size?: 'normal' | 'small';
    className?: string;
    autoFocus?: boolean;
    disabled?: boolean;
    name?: string;
    type: 'submit' | 'reset' | 'button';
    value?: string | ReadonlyArray<string> | number;
    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
};

export default function Button (props: PropsWithChildren<ButtonProps>): ReactElement {
    const buttonClasses = ['btn', `btn--${props.mode}`];

    if (props.disabled) {
        buttonClasses.push('btn--disabled');
    }

    if (props.size === 'small') {
        buttonClasses.push('btn--small');
    }

    return (
        <button {...props}
            className={`${buttonClasses.join(' ')} ${props.className || ''}`}>
            {props.children}
        </button>
    );
}
