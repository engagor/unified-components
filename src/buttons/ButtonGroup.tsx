import React, { PropsWithChildren, ReactElement } from "react";

export type ButtonGroupProps = {
    className?: string;
};

export default function ButtonGroup (props: PropsWithChildren<ButtonGroupProps>): ReactElement {
    return (
        <div className={`d-flex ${props.className || ''}`}>
            {props.children}
        </div>
    );
}
