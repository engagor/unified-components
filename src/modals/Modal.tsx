import '@clarabridge/unified-ui-tailwind/dist/modals.css';
import React, { PropsWithChildren, ReactElement } from 'react';

export type ModalProps = PropsWithChildren<
    {
        size?: 'large' | 'medium' | 'small'
        'aria-labelledby'?: string;
        'aria-described-by'?: string;
    }
>;

export default function Modal(props: ModalProps): ReactElement {
    const size = props.size ?? 'medium';

    return (
        <dialog
            className={`modal modal--${size}`}
            aria-labelledby={props['aria-labelledby']}
            aria-describedby={props['aria-described-by']}
        >
            {props.children}
        </dialog>
    );
}
