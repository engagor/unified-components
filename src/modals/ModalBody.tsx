import '@clarabridge/unified-ui/dist/modals.css';
import React, { PropsWithChildren, ReactElement } from 'react';

export type ModalBodyProps = PropsWithChildren<
    {

    }
>;

export default function ModalBody(props: ModalBodyProps): ReactElement {
    return <section className="modal__body">{props.children}</section>;
}
