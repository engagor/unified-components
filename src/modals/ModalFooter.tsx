import '@clarabridge/unified-ui/dist/modals.css';
import React, { PropsWithChildren, ReactElement } from 'react';

export type ModalFooterProps  = PropsWithChildren<
    {

    }
>;

export default function ModalFooter(props: ModalFooterProps): ReactElement {
    return <footer className="modal__footer">{props.children}</footer>;
}
