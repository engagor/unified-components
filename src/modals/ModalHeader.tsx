import '@clarabridge/unified-ui/dist/modals.css';
import React, { PropsWithChildren, ReactElement, MouseEvent } from 'react';

export type ModalHeaderProps  = PropsWithChildren<
    {
        mode?: 'default' | 'warning' | 'danger';
        onClose?: (event: MouseEvent<HTMLButtonElement>) => void;
    }
>;

export default function ModalHeader(props: ModalHeaderProps): ReactElement {
    const mode = props.mode ?? 'default';

    return (
        <header className={`modal__header modal__header--${mode}`}>
            <h2 id="modal-header" className="modal__header__title">{props.children}</h2>
            <button className="modal__header__dismiss" type="button" aria-label="Close modal" onClick={props.onClose}>
                <span className="en-icon en-icon-delete-3" aria-hidden="true" />
            </button>
        </header>
    );
}
