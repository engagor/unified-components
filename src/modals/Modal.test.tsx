import React  from 'react';
import renderer from 'react-test-renderer';
import Modal from './Modal';
import ModalHeader from './ModalHeader';
import ModalBody from './ModalBody';
import ModalFooter from './ModalFooter';
import ButtonGroup from '../buttons/ButtonGroup';
import Button from '../buttons/Button';

it('renders correctly', async () => {
    const modal = await renderer
        .create(
            <Modal size="medium" aria-labelledby="modal-header">
                <ModalHeader mode="default" onClose={() => { /** Do nothing */ }}>Modal header</ModalHeader>
                <ModalBody>
                    <p className="text-black dark:text-white m-0">Here comes the body</p>
                </ModalBody>
                <ModalFooter>
                    <ButtonGroup className="justify-end w-100-percent">
                        <Button mode="secondary" type="button" className="mr-8">Cancel</Button>
                        <Button mode="primary" type="button">Save</Button>
                    </ButtonGroup>
                </ModalFooter>
            </Modal>
        )
        .toJSON();

    expect(modal).toMatchSnapshot();
});
