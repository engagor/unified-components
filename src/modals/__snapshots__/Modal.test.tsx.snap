// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<dialog
  aria-labelledby="modal-header"
  className="modal modal--medium"
>
  <header
    className="modal__header modal__header--default"
  >
    <h2
      className="modal__header__title"
    >
      Modal header
    </h2>
    <button
      aria-label="Close modal"
      className="modal__header__dismiss"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
        className="en-icon en-icon-delete-3"
      />
    </button>
  </header>
  <section
    className="modal__body"
  >
    <p
      className="text-black dark:text-white m-0"
    >
      Here comes the body
    </p>
  </section>
  <footer
    className="modal__footer"
  >
    <div
      className="flex justify-end w-100-percent"
    >
      <button
        className="btn btn--secondary mr-8"
        mode="secondary"
        type="button"
      >
        Cancel
      </button>
      <button
        className="btn btn--primary "
        mode="primary"
        type="button"
      >
        Save
      </button>
    </div>
  </footer>
</dialog>
`;
